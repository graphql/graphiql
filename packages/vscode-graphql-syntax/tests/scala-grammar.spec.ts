import { tokenizeFile } from './__utilities__/utilities';

describe('inline.graphql.scala grammar', () => {
  const scope = 'inline.graphql.scala';

  it('should tokenize a simple scala file', async () => {
    const result = await tokenizeFile('__fixtures__/test.scala', scope);
    expect(result).toMatchSnapshot();
  });
});
