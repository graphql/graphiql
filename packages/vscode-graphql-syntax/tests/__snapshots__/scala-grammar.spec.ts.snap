// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`inline.graphql.scala grammar > should tokenize a simple scala file 1`] = `
val query =   | 
graphql       | keyword.interpolation.scala
"""           | string.quoted.triple.scala
              | meta.embedded.block.graphql meta.selectionset.graphql
{             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
id            | meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
}             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
"""           | string.quoted.triple.scala
val query2 =  | 
gql           | keyword.interpolation.scala
"""           | string.quoted.triple.scala
              | meta.embedded.block.graphql meta.selectionset.graphql
{             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
id            | meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
}             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
"""           | string.quoted.triple.scala
val query3 =  | 
schema        | keyword.interpolation.scala
"""           | string.quoted.triple.scala
              | meta.embedded.block.graphql meta.selectionset.graphql
{             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
id            | meta.embedded.block.graphql meta.selectionset.graphql variable.graphql
              | meta.embedded.block.graphql meta.selectionset.graphql
}             | meta.embedded.block.graphql meta.selectionset.graphql punctuation.operation.graphql
"""           | string.quoted.triple.scala
              | 
val query =   | 
"""           | string.quoted.triple.scala
#graphql      | comment.line.graphql.js
              | meta.selectionset.graphql
{             | meta.selectionset.graphql punctuation.operation.graphql
              | meta.selectionset.graphql
id            | meta.selectionset.graphql variable.graphql
              | meta.selectionset.graphql
}             | meta.selectionset.graphql punctuation.operation.graphql
"""           | string.quoted.triple.scala
              | 
`;
